homeassistant/components/peco/config_flow.py

```diff
@@ -1,34 +1,80 @@
 """Config flow for PECO Outage Counter integration."""
 from __future__ import annotations
 
+import asyncio
 from typing import Any
 
+from peco import (
+    HttpError,
+    IncompatibleMeterError,
+    PecoOutageApi,
+    UnresponsiveMeterError,
+)
 import voluptuous as vol
 
 from homeassistant import config_entries
 from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers import config_validation as cv
 
-from .const import CONF_COUNTY, COUNTY_LIST, DOMAIN
+from .const import CONF_COUNTY, CONF_PHONE_NUMBER, COUNTY_LIST, DOMAIN
 
-STEP_USER_DATA_SCHEMA = vol.Schema(
+STEP_OUTAGE_COUNTER_DATA_SCHEMA = vol.Schema(
     {
         vol.Required(CONF_COUNTY): vol.In(COUNTY_LIST),
     }
 )
+STEP_SMART_METER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_PHONE_NUMBER): cv.string,
+    }
+)
 
 
 class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
     """Handle a config flow for PECO Outage Counter."""
 
     VERSION = 1
 
+    meter_verification: bool = False
+    meter_data: dict[str, str] = {}
+    meter_error: dict[str, str] = {}
+
+    async def _verify_meter(self, phone_number: str) -> None:
+        """Verify if the meter is compatible."""
+
+        api = PecoOutageApi()
+
+        try:
+            await api.meter_check(phone_number)
+        except ValueError:
+            self.meter_error = {"base": "invalid_phone_number", "type": "error"}
+        except IncompatibleMeterError:
+            self.meter_error = {"base": "incompatible_meter", "type": "abort"}
+        except UnresponsiveMeterError:
+            self.meter_error = {"base": "unresponsive_meter", "type": "error"}
+        except HttpError:
+            self.meter_error = {"base": "http_error", "type": "error"}
+
+        self.hass.async_create_task(
+            self.hass.config_entries.flow.async_configure(flow_id=self.flow_id)
+        )
+
     async def async_step_user(
         self, user_input: dict[str, Any] | None = None
     ) -> FlowResult:
         """Handle the initial step."""
+        return self.async_show_menu(
+            step_id="user", menu_options=["outage_counter", "smart_meter"]
+        )
```

Comment:
Why would those be different? Why can't a user just set up both in one go instead of having to set up the same integration multiple times?

Or should these have been different integrations?

Reply:
I could have the user set both up in one go. Would that be better?

P.S. Built in PR reviews now?! So cool! No more alt tabbing!
![A screenshot of VS Code showing a popup over the code editor displaying the contents of a review comment](https://ipradio.app/cdn/hTbHp.png)

Reply:
> Would that be better?
> 

Yes